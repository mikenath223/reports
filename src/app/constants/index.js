export const pieColors = ["#A259FF", "#F24E1E", "#FFC107", "#6497B1"];
