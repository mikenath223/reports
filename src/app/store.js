import { configureStore } from "@reduxjs/toolkit";
import logger from "redux-logger";
import projectsReducer from "./slices/projectsSlice";
import formReducer from "./slices/formSlice";
import userReducer from "./slices/userSlice";
import gatewayReducer from "./slices/gatewaySlice";
import reportsReducer from "./slices/reportsSlice";

const middlewares = [];
if (process.env.NODE_ENV === `development`) {
  middlewares.push(logger);
}

export const store = configureStore({
  reducer: {
    projects: projectsReducer,
    form: formReducer,
    users: userReducer,
    gateways: gatewayReducer,
    reports: reportsReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(middlewares),
});
