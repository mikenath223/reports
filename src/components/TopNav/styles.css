.top-nav {
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 1.5rem 5rem 1.5rem 2rem;
  border-bottom: 2px solid #f3f6f9;
}

.initials {
  text-align: center;
  color: #fff;
  background-color: #f6ca65;
  border-radius: 5px;
  padding: 10px 10px;
  font-size: 23px;
  font-weight: 700;
}

.flex-wrap {
  display: flex;
  align-items: center;
  gap: 20px;
}

.logo-gap {
  gap: 40px;
}

.username {
  color: #005b96;
  font-weight: 700;
  font-size: 16px;
}
