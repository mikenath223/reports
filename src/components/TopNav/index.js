import { ReactComponent as Logo } from "app/images/logo1.svg";
import { ReactComponent as MenuIcon } from "app/images/menu.svg";
import { getUsers } from "app/slices/userSlice";
import { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import "./styles.css";

export default function TopNav() {
  const dispatch = useDispatch();
  const user = useSelector((state) => state.users.data);
  const firstUser = user?.[0];
  const userInititals = `${firstUser?.firstName?.at(0)?.toUpperCase() || ""}${
    firstUser?.lastName?.at(-1)?.toUpperCase() || ""
  }`;
  const userName = `${firstUser?.firstName || ""} ${firstUser?.lastName || ""}`;

  useEffect(() => {
    dispatch(getUsers());
  }, [dispatch]);

  return (
    <header className="top-nav">
      <div className="flex-wrap logo-gap">
        <Logo className="" />
        <MenuIcon />
      </div>
      <div className="flex-wrap">
        <h1 className="initials">{userInititals}</h1>
        <span className="username">{userName}</span>
      </div>
    </header>
  );
}
