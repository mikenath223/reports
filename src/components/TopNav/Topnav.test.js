import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { store } from "app/store";
import TopNav from ".";

describe("Topnav", function () {
  it("should match snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <TopNav />
      </Provider>
    );

    expect(container).toMatchSnapshot();
  });
});
