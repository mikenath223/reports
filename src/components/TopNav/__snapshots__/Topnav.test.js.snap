// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topnav should match snapshot 1`] = `
<div>
  <header
    class="top-nav"
  >
    <div
      class="flex-wrap logo-gap"
    >
      <svg
        class=""
      >
        logo1.svg
      </svg>
      <svg>
        menu.svg
      </svg>
    </div>
    <div
      class="flex-wrap"
    >
      <h1
        class="initials"
      />
      <span
        class="username"
      >
         
      </span>
    </div>
  </header>
</div>
`;
